src/adapter/source-adapter.ts

```diff
@@ -11,4 +11,6 @@ export interface SourceAdapter<C, L, A> extends Adapter {
   getAllArticles(): Promise<A[]>;
 
   getDocumentLinkMatcherRegexp(): RegExp | undefined;
+
+  getBaseUrl(): string;
```

Comment:
Won't this name be confusing later on? This function returns config.relativeLinkBaseUrl with a fallback to the baseUrl returned by service apis.

Reply:
It can be getResourceBaseUrl